render/egl.c

```diff
@@ -358,6 +358,58 @@ struct wlr_egl *wlr_egl_create(EGLenum platform, void *remote_display,
 	return NULL;
 }
 
+struct wlr_egl *wlr_egl_create_from_drm_fd(int drm_fd) {
+	PFNEGLQUERYDEVICESTRINGEXTPROC eglQueryDeviceStringEXT;
+	PFNEGLQUERYDEVICESEXTPROC eglQueryDevicesEXT;
+
+	load_egl_proc(&eglQueryDevicesEXT, "eglQueryDevicesEXT");
+	load_egl_proc(&eglQueryDeviceStringEXT,"eglQueryDeviceStringEXT");
```

Comment:
We need to check extension strings before doing that. EGL might return garbage function pointers otherwise.

Reply:
I thought it was a "load or die" function https://github.com/swaywm/wlroots/blob/master/render/egl.c#L91
I'll pull the exts list before loading then.

Reply:
Yeah, EGL doesn't guarantee that sadly. The assertions are just in case the EGL impl is nice enough.